torch/testing/_internal/distributed/rpc/rpc_test.py

```diff
@@ -4736,6 +4762,33 @@ def test_rpc_script_timeout(self):
 
 class TensorPipeAgentRpcTest(RpcAgentTestFixture):
 
+    @skip_if_lt_x_gpu(2)
+    def test_future_extraction_sparse_tensor(self):
+        try:
+            opts = rpc.TensorPipeRpcBackendOptions(init_method=self.init_method)
+            if self.rank == 1:
+                rpc.init_rpc(
+                    "master",
+                    rank=self.rank,
+                    world_size=2,
+                    rpc_backend_options=opts
+                )
+                rref = rpc.remote("server", Server)
+                tensor = build_sparse_tensor()
+                tensor = sparse_tensor_to_rpc_format(tensor)
+                fut = rref.rpc_async().identity(tensor).then(sparse_callback)
+                fut.wait()
+            elif self.rank == 0:
+                rpc.init_rpc(
+                    "server",
+                    rank=self.rank,
+                    world_size=2,
+                    rpc_backend_options=opts
+                )
+        finally:
+            if self.rank < 2:
+                rpc.shutdown()
```

Comment:
This test won't actually do anything, because DataPtr extraction only occurs for CUDA-enabled Futures, whereas here we're CPU-only.

Reply:
My misunderstanding will remove.